cpp/arcticdb/version/version_map.hpp

```diff
@@ -564,6 +566,25 @@ class VersionMapImpl {
             entry->validate();
     }
 
+    void verify_stream_id(const std::shared_ptr<Store> &store,
+                   const StreamId &stream_id) {
+        if (has_cached_entry(stream_id, LoadParameter{LoadType::LOAD_LATEST}) || has_stored_entry(store, stream_id))
+            return;
+
+        if (std::getenv("ARCTICDB_NO_STRICT_SYMBOL_CHECK"))
+        {
+            ARCTICDB_DEBUG(log::version(), "Key with stream id {} will not be strictly checked because ARCTICDB_NO_STRICT_SYMBOL_CHECK env var is set.", stream_id);
+            return;
+        }
+
+        for (unsigned char c : std::get<std::string>(stream_id)) {
+            if (c < 32 || c > 127) {
+                throw UserInputException(
```

Comment:
Yeah, we should add an error code for this (a new User Input error). See `error_messages.md` and have a look at how we implement them in `error_code.hpp`. Use `user_input::raise` to raise it.